build_tools/create_log_index.py

```diff
@@ -21,22 +21,57 @@ def normalize_path(p: Path) -> str:
     return str(p).replace("\\", "/") if is_windows() else str(p)
 
 
+def get_indexer_path() -> Path:
+    indexer_env_path = os.getenv("INDEXER_PATH")
+    if indexer_env_path:
+        indexer_path = Path(indexer_env_path)
+        if indexer_path.is_file():
+            log(f"[INFO] Using INDEXER_PATH from environment: {indexer_path}")
+            return indexer_path
+        else:
+            log(
+                f"[ERROR] INDEXER_PATH is set but does not point to a file: {indexer_path}"
+            )
+            log(f"[INFO] CWD is: {os.getcwd()}")
+            log(f"[INFO] Contents of dirname: {list(indexer_path.parent.glob('*'))}")
+            sys.exit(2)
+
+    fallback_path = Path("build/indexer.py")
+    if fallback_path.is_file():
+        log(f"[INFO] Falling back to default indexer path: {fallback_path}")
+        return fallback_path
+
+    log(
+        "[ERROR] INDEXER_PATH is not set and fallback path does not exist. Cannot continue."
+    )
+    sys.exit(2)
+
+
 def index_log_files(build_dir: Path, amdgpu_family: str):
     log_dir = build_dir / "logs"
```

Comment:
Here the script should be given the path to the logs directory and shouldn't need to assume that logs are always in `${build_dir}/logs`

Reply:
@ScottTodd, How can I find out the log path?

Reply:
CMake always sets to `${THEROCK_BINARY_DIR}/logs`, so that is safe to assume: https://github.com/ROCm/TheRock/blob/46ba6a0b37ba7abd2f3cbe43cfe995f170bd15b3/cmake/therock_subproject.cmake#L118-L121

This script looks like it only cares about the logs directory though, so it should be passed only the path to the logs directory, not the build directory.